### PR TITLE
Android platforms announcement

### PR DESCRIPTION
**Pull request by katre**

### CODE REVIEW

**Comment by katre** [comment]
Ideally I'd like to publish this on Friday, November 17. I can change the date of the file, but do I also need to hold back merging this PR until that date?

File: _posts/2023-11-13-android-platforms.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+Just like the `--platforms` flag, the values passed to `--android_platforms` are
+the labels of [`platform`](https://bazel.build/reference/be/platforms-and-toolchains#platform) targets, using standard constraint values to describe your device.
+
+For example, for an Android device with a 64-bit ARM processor, you'd define
+your platform like this:
+
+```py
+platform(
+    name = "arm64",
```
Comment (meisterT) [inline _posts/2023-11-13-android-platforms.md:26]: Should we name this `android_arm64`?

File: _posts/2023-11-13-android-platforms.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  used in.
+- `--android_crosstool_top`: Similarly to the Android SDK, the
+  `android_ndk_repository` rule exposes the NDK as a C++ toolchain, using the
+  standard toolchain resolution. You can use `register_toolchains` and
+  `--extra_toolchains` to change the selection order for these.
+
+### Can I test this out?
+
+Yes, download the latest [Bazel 7.0 release
+candidate](https://releases.bazel.build/7.0.0/rc3/index.html) and update your
```
Comment (meisterT) [inline _posts/2023-11-13-android-platforms.md:67]: Nit: We plan to cut a new release candidate likely tomorrow.

Perhaps we should mention that with bazelisk / `last_rc` you get the most recent RC?

File: _posts/2023-11-13-android-platforms.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+At a more basic level, because other rules (like C++, Java, Go, and Rust) are
+moving to support platforms, Android needs to support it to, so that
+mixed-language builds work properly.
+
+## What's next for Android Rules?
+
+The biggest change coming to the Android rules is to move them out of the Bazel
+binary, and into [the independent `rules_android`
+repository](https://github.com/bazelbuild/rules_android). Starlarkification of
+the Android Rules is expected to be complete in Q2 2024, and users will be
```
Comment (meisterT) [inline _posts/2023-11-13-android-platforms.md:108]: Consider linking to https://github.com/orgs/bazelbuild/projects/17

**Comment by meisterT** [comment]
> Ideally I'd like to publish this on Friday, November 17. I can change the date of the file, but do I also need to hold back merging this PR until that date?

Yes, do not merge if you don't want it to go live.

File: _posts/2023-11-13-android-platforms.md (hunk 2 above)
Comment (katre) [reply _posts/2023-11-13-android-platforms.md]: Too bad we don't have a central page for release candidates. Added a link for Bazelisk.

File: _posts/2023-11-13-android-platforms.md (hunk 1 above)
Comment (katre) [reply _posts/2023-11-13-android-platforms.md]: Makes sense

File: _posts/2023-11-13-android-platforms.md (hunk 2 above)
Comment (meisterT) [reply _posts/2023-11-13-android-platforms.md]: cc @fweikert as we talked about this earlier

**Comment by katre** [comment]
Current plan is to merge this on Friday, November 17, but I'm happy to keep taking suggestions until there.

File: _posts/2023-11-17-android-platforms.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+As part of the overall [platforms and
+toolchains](https://bazel.build/extending/platforms) project, all of Bazel is
+moving away from using `--cpu` flags to express target platforms, and towards
+the richer variety allowed by the `platform` rule. The current usage of `--cpu`
+is not actually as a single CPU architecture: we currently capture different
+things like "Linux on 64-bit intel" (`--cpu=k8`), "macOS on 64-bit-intel"
+(`--cpu=darwin_x86_64`), "Windows on ARM" (`--cpu=arm64_windows`): and that's
+just a few common OSes.
+
+Once we also want to describe Android and iOS devices, or custom development
```
Comment (aiuto) [inline _posts/2023-11-17-android-platforms.md:96]: "Once we also want to ..." seems a little awkward.

File: _posts/2023-11-17-android-platforms.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+The biggest change coming to the Android rules is to move them out of the Bazel
+binary, and into [the independent `rules_android`
+repository](https://github.com/bazelbuild/rules_android). [Starlarkification of
+the Android Rules](https://github.com/orgs/bazelbuild/projects/17) is expected
+to be complete in Q2 2024, and users will be encouraged to migrate to the
+Starlark Android Rules between then and Bazel 8.0
+
+The current version of the Starlark Android rules also supports the same
+`--android_platforms` flag, although since these rules aren't currently fully
+ready for release, there will be breaking issues.
```
Comment (aiuto) [inline _posts/2023-11-17-android-platforms.md:115]: "there will be breaking issues" raises the question, what issues?

File: _posts/2023-11-17-android-platforms.md (hunk 4 above)
Comment (katre) [reply _posts/2023-11-17-android-platforms.md:96]: Done.

File: _posts/2023-11-17-android-platforms.md (hunk 5 above)
Comment (katre) [reply _posts/2023-11-17-android-platforms.md]: How about this?

**Review by aiuto — APPROVED** [review]

**Comment by katre** [comment]
Actually, if this is ready to go, I'd like to move it up and publish today. Anyone have an objection?

**Comment by meisterT** [comment]
SGTM

**Pull request merged**